Clang Formatting

Skptak opened the pull request:
Add in a CI-CD action to clang-format files, as well as check for CRLF line endings and trailing whitespace
Add relevant test cases as well.

kstribrnAmzn commented:
Couple of things:
1. I'd love to see smaller example files used. The example files alone for this PR are MASSIVE. In total this PR is adding roughly 60k lines. It's not exactly easy to review.
2. It would be nice to have a comment block at the top of the example files explaining how exactly they were generated and what the problem with the file is. While the names are nice, I can't imagine that you change 50k+ lines for formatting failures

kstribrnAmzn commented:
Approving because I don't see any stand out problems but I obviously haven't read every single line in depth.

Skptak commented:
> Couple of things:
> 
>     1. I'd love to see smaller example files used. The example files alone for this PR are MASSIVE. In total this PR is adding roughly 60k lines. It's not exactly easy to review.

Replaced the files with short ones that are 200 lines long at most.
It was an intentional decision to select files from various FreeRTOS Repos:
1. To show that the formatting actions then also work for files we have.
2. To make my life easier by not needing to generate C or H files that had issues, as I just grabbed files that failed checks when implementing the clang-format action.
Additionally, certain FreeRTOS Files will fail to parse through uncrustify using uncrustify version .72. Where the idea was to grab long ones so that if you're testing the action itself we see that these long files will handle the new formatting changes, if one was changing the .clang-format file for example.
But not going to hold up this PR about this since both you and @paulbartell  have requested shorter files.
 
>     2. It would be nice to have a comment block at the top of the example files explaining how exactly they were generated and what the problem with the file is. While the names are nice, I can't imagine that you change 50k+ lines for formatting failures
Do you still feel this way with the current naming convention?
I have also added a README explaining the purpose of these files.

bradleysmith23 commented on clang-formatting/filesWithFormattingErrors/goodFileInHere.c:
```diff
@@ -0,0 +1,100 @@
+    uint64_t minutes;
+    uint64_t seconds;
+    uint64_t msec;
+} DateAndTime;
```
Should this be changed from DateAndTime? I do not see anything related to the date

Skptak replied:
Entirely a fake C file that is just used to catch formatting errors, any content in the file itself related to code is irrelevant.

bradleysmith23 commented on clang-formatting/filesWithFormattingErrors/testWithFormattingError.c:
```diff
@@ -0,0 +1,95 @@
+    uint64_t minutes;
+    uint64_t seconds;
+    uint64_t msec;
+} DateAndTime;
```
Same comment as [here](https://github.com/FreeRTOS/CI-CD-Github-Actions/pull/82/files#r1316479587)

bradleysmith23 commented on clang-formatting/filesWithTrailingWhitespace/testWithTrailingWhitespace.c:
```diff
@@ -0,0 +1,94 @@
+    uint64_t minutes;
+    uint64_t seconds;
+    uint64_t msec;
+} DateAndTime;
```
Same comment as [here](https://github.com/FreeRTOS/CI-CD-Github-Actions/pull/82/files#r1316479587)

Skptak replied:
Entirely a fake C file that is just used to catch formatting errors, any content in the file itself related to code is irrelevant.

bradleysmith23 commented on clang-formatting/goodFiles/source/README.md:
```diff
@@ -0,0 +1,100 @@
+    uint64_t minutes;
+    uint64_t seconds;
+    uint64_t msec;
+} DateAndTime;
```
Same comment as [here](https://github.com/FreeRTOS/CI-CD-Github-Actions/pull/82/files#r1316479587)

Skptak replied:
Entirely a fake C file that is just used to catch formatting errors, any content in the file itself related to code is irrelevant.

bradleysmith23 commented on clang-formatting/goodFiles/source/fileWithErrorSource/errorFileInDirectory.c:
```diff
@@ -0,0 +1,108 @@
+    uint64_t minutes;
+    uint64_t seconds;
+    uint64_t msec;
+} DateAndTime;
```
Same comment as [here](https://github.com/FreeRTOS/CI-CD-Github-Actions/pull/82/files#r1316479587)

Skptak replied:
Entirely a fake C file that is just used to catch formatting errors, any content in the file itself related to code is irrelevant.

bradleysmith23 commented on clang-formatting/goodFiles/include/formatErrorTest.h:
```diff
@@ -0,0 +1,49 @@
+    #error "FCLK_FREQUENCY Error: Please enter a valid divider value"
+#endif
+
+#if(                        ICLK_FREQUENCY                                      >                            100000000L )
```
remove extra spaces

Skptak replied:
Entirely a fake C file that is just used to catch formatting errors, any content in the file itself related to code is irrelevant.

bradleysmith23 commented on clang-formatting/goodFiles/include/formatErrorTest.h:
```diff
@@ -0,0 +1,49 @@
+#endif
+
+#if( CLK_SOURCE == CLK_SOURCE_HOCO ) && ( ENABLE_HOCO == 0 )
+    #error "HOCO has been specified as the CLK_SOURCE but           dsafa                                            ENABLE_HOCO is (0). Please set to (1) in file cgc.h"
```
remove extra spaces
